web/source/osk/oskManager.ts

```diff
@@ -482,22 +470,40 @@ namespace com.keyman.osk {
     }
 
     /**
-     * Move OSK back to default position
+     * Function     restorePosition
+     * Scope        Public
+     * @param       {boolean?}      keepDefaultPosition  If true, does not reset the default x,y set by `setRect`.
+     *                                                   If false or omitted, resets the default x,y as well.
+     * Description  Move OSK back to default position, floating under active input element
      */
-    restorePosition: () => void = function(this: OSKManager) {
-      if(this._Visible) {
+    ['restorePosition']: (keepDefaultPosition?: boolean) => void = function(this: OSKManager, keepDefaultPosition?: boolean) {
+      let isVisible = this._Visible;
+      if(isVisible) {
         com.keyman.singleton.domManager.focusLastActiveElement();  // I2036 - OSK does not unpin to correct location
-        this.loadCookie();
-        this.userPositioned=false;
-        this.saveCookie();
+      }
+
+      this.loadCookie();
+      this.userPositioned=false;
+      if(!keepDefaultPosition) {
+        delete this.dfltX;
+        delete this.dfltY;
```

Comment:
Resets these values only if called by API consumer (with falsy parameter) but not by pin div onclick...